app/models/illiad_request.rb

```diff
@@ -0,0 +1,87 @@
+###
+#  Class to handle creation of ILLiad OpenURL request
+###
+class IlliadRequest
+  def initialize(current_user, scan)
+    @current_user = current_user
+    @scan = scan
+  end
+
+  def illiad_transaction_response
+    response.respond_to?(:body) ? response.body : response
```

Comment:
Is there a case where the `response` object is something we want to be dealing with instead of the body of the response?